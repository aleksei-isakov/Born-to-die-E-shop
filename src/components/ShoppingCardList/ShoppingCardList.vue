<template>
  <div class="cart__wrapper">
    <div class="shopping-cart__wrapper">
      <shopping-card-item
        v-for="(cartItem, i) in shoppingCartData"
        :key="i"
        :description="cartItem.description"
        :quantity="cartItem.quantity"
        :price="cartItem.price"
        :index="i"
        v-on="$listeners"
      />
    </div>
    <div class="total-price">{{ totalPrice }}</div>
  </div>
</template>

<script>
import ShoppingCardItem from '@/components/ShoppingCardItem/ShoppingCardItem';

export default {
  name: 'ShoppingCardList',

  components: { ShoppingCardItem },

  props: {
    shoppingCartData: {
      type: Array,
      default: () => [],
      require: true
    }
  },

  computed: {
    totalPrice() {
      let totalPrice = 0;

      if (this.shoppingCartData.length === 0) return `Total Price: 0 $`;
      totalPrice = this.shoppingCartData.map(({ price, quantity }) => {
        return price * quantity;
      });

      return `Total Price: ${totalPrice.reduce((acc, _) => acc + _)} $`;
    }
  }
};
</script>

<style lang="scss" scoped>
@import '@/scss/CustomVariables.scss';

.shopping-cart__wrapper {
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  gap: 20px;
}

.total-price {
  margin-right: 50rem;
  margin-top: 40px;
  color: $font-color-subtitle;
  font-size: 1.1rem;
}

.cart__wrapper {
  display: flex;
  flex-direction: column;
}

@media screen and (max-width: $tablet-size) {
  .shopping-cart__wrapper {
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
    gap: 10px;
  }

  .total-price {
    width: 150px;
  }

  .cart__wrapper {
    display: flex;
    flex-direction: column;
  }
}
</style>
