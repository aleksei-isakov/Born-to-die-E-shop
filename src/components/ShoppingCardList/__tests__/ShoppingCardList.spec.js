import { mount } from '@vue/test-utils';
import ShoppingCardList from '@/components/ShoppingCardList/ShoppingCardList';

let wrapper;

beforeEach(() => {
  wrapper = mount(ShoppingCardList);
});

afterEach(() => {
  wrapper.destroy();
});

describe('PhotoSlider', () => {
  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance).toBeTruthy();
  });
  test('match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
