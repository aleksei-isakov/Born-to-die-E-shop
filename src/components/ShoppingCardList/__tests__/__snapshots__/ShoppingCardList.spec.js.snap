// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoSlider match snapshot 1`] = `
<div class="cart__wrapper">
  <div class="shopping-cart__wrapper">
    <div class="shoppingCard__wrapper"><img src="" class="cart-item-img">
      <div class="shopping-cart-item-description">description</div>
      <div class="product-price">35 $</div>
      <div class="quantity__wrapper"><button class="base-button base-text-button base-text-filled-button blue">-</button> <input readonly="readonly" class="quantity-field"> <button class="base-button base-text-button base-text-filled-button blue">+</button></div>
      <div>
        <v-icon class="delete-icon">fas fa-trash</v-icon>
      </div>
    </div>
    <div class="shoppingCard__wrapper"><img src="" class="cart-item-img">
      <div class="shopping-cart-item-description">description</div>
      <div class="product-price">36 $</div>
      <div class="quantity__wrapper"><button class="base-button base-text-button base-text-filled-button blue">-</button> <input readonly="readonly" class="quantity-field"> <button class="base-button base-text-button base-text-filled-button blue">+</button></div>
      <div>
        <v-icon class="delete-icon">fas fa-trash</v-icon>
      </div>
    </div>
    <div class="shoppingCard__wrapper"><img src="" class="cart-item-img">
      <div class="shopping-cart-item-description">description</div>
      <div class="product-price">37 $</div>
      <div class="quantity__wrapper"><button class="base-button base-text-button base-text-filled-button blue">-</button> <input readonly="readonly" class="quantity-field"> <button class="base-button base-text-button base-text-filled-button blue">+</button></div>
      <div>
        <v-icon class="delete-icon">fas fa-trash</v-icon>
      </div>
    </div>
    <div class="shoppingCard__wrapper"><img src="" class="cart-item-img">
      <div class="shopping-cart-item-description">description</div>
      <div class="product-price">38 $</div>
      <div class="quantity__wrapper"><button class="base-button base-text-button base-text-filled-button blue">-</button> <input readonly="readonly" class="quantity-field"> <button class="base-button base-text-button base-text-filled-button blue">+</button></div>
      <div>
        <v-icon class="delete-icon">fas fa-trash</v-icon>
      </div>
    </div>
  </div>
  <div class="total-price">Total Price: 146 $</div>
</div>
`;
